### ISSUE TITLE
Response event is not intercepter using Supertest

### ISSUE THREAD

**Issue by djom202** [mswjs/interceptors#277]
- Dependencies:
```
"node": "v16.15.1",
"supertest": "^6.1.6",
"jest": "^28.1.3",
"@mswjs/interceptors": "^0.17.3",
```

- Code:
```javascript
import { app } from '../base/config'
import { ClientRequestInterceptor } from '@mswjs/interceptors/lib/interceptors/ClientRequest'
const interceptor = new ClientRequestInterceptor()

interceptor.on('request', async (request) => {
  console.log(request.method, request.url.href)
  console.log('request', request)
})

interceptor.on('response', (response, request) => {
  console.log(
    'response to %s %s was:',
    request.method,
    request.url.href,
    response
  )
})

beforeEach(() => {
  interceptor.apply()
})

after(() => {
  interceptor.dispose()
})

describe("Sanity tests", () => {
  test("should return all options", async () => {
    const response = await app
      .get('v1/options')
      .set(headers.common)
 
    allure.createAttachment('Response', JSON.stringify(response.body))
  })
})
```

- issue description:
I used the code like you mention in the README.md and the request event works fine as expected, nevertheless the response event never is called. Also I'm not sure if maybe the way to use the supertest or if something that I'm missing in the code.

Also this example doesn't work too with the response event: https://github.com/mswjs/interceptors/blob/main/test/third-party/supertest.test.ts

**Comment by kettanaito** [comment]
Hey, @djom202. Thanks for reporting this.

Could you please give this another try with the latest version of the library (0.18.2 at the moment). It does contain some breaking changes to how requests and responses are represented but also ships a ton of improvements all over the place. I suspect this issue is no more. I'd be thankful if you confirmed this.

**Comment by kettanaito** [comment]
I've just added automated tests to ensure that both "request" and "response" events are emitted when using Supertest. No additional changes to the library were required, it works as expected on the latest version. 

On the relevant note, I can see a few issues with your setup:

1. Do not call `interceptor.apply()` before every test; instead, call it `beforeAll()`.
2. The `after` hook doesn't exist in Jest, so it's never called (it ought to throw). It should be `afterAll()` instead.

**Comment by kettanaito** [comment]
## Released: v0.19.1 🎉

This has been released in v0.19.1!

- 📄 [**Release notes**](https://github.com/mswjs/interceptors/releases/tag/v0.19.1)
- 📦 [npm package](https://www.npmjs.com/package/@mswjs/interceptors/v/0.19.1)

Make sure to always update to the latest version (`npm i @mswjs/interceptors@latest`) to get the newest features and bug fixes.

---

_Predictable release automation by [@ossjs/release](https://github.com/ossjs/release)_.
